components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarAutocompleteFeature.kt

```diff
@@ -40,32 +42,10 @@ class ToolbarAutocompleteFeature(val toolbar: Toolbar) {
         domainProviders.add(provider)
     }
 
-    // The rest is a little bit of boilerplate to tie things together.
-    // We terminate our dependency chains at the 'feature' level. This is an alternative to creating
-    // an "AutocompleteProvider" concept. This pattern allows a storage component, for example, to
```

Comment:
From this comment here, I'm not sure if the intent was to write an `AutocompleteProvider` concept eventually or this was the alternative to it.

Reply:
HistoryStorage may have been built concurrently while the AwesomeBar was being built so that may have been the reasoning for the different APIs when it comes to retrieving the autocomplete results.